lib/rucio/core/did.py

```diff
@@ -2845,109 +2843,12 @@ def set_status(scope, name, *, session: "Session", **kwargs):
                 rucio.core.rule.generate_rule_notifications(rule=rule, session=session)
 
 
-@stream_session
+@read_session
```

Comment:
I think this should still be a `stream_session`, simply because list_dids in the meta_plugin is also a `stream_session`.

Reply:
If I remember correctly, the only reason I changed this was because it gave me an error telling me to do so .

Reply:
`Details: stream_session decorator should be used only with generator. Use read_session instead.` 

looks like there's a mixture of returns and yields in the list_dids chain i.e. from core->did.py through core->did_meta_plugins->\_\_init\_\_.py to the corresponding plugin module.

I'll leave it as-is for now (as it doesn't work otherwise)?